feat(shutter): add Shutter DAO 0x36 integration guide and tests

0xferit opened the pull request:
## Summary

Deployment playbook and integration tests for Shutter DAO 0x36 to deploy a Dragon Vault with 1.2M USDC via their Fractal/Decent governance.

Includes prepared calldata for all DAO proposal transactions and gas verification under the 16.7M per-tx limit.

## References

- [Pilot Proposal (Forum)](https://shutternetwork.discourse.group/t/octant-v2-pilot-to-optimize-treasury-strengthen-ecosystem/760)
- [Treasury](https://etherscan.io/address/0x36bD3044ab68f600f6d3e081056F34f2a58432c4)

0xferit committed:
feat(shutter): add Shutter DAO 0x36 integration guide and tests

qGolem commented on docs/shutter_dao_integration.md:
```diff
@@ -0,0 +1,322 @@
+
+Shutter DAO 0x36 will integrate with Octant v2 through **two distinct components**:
+
+1. **Dragon Vault (MSLV)** — A Multistrategy Locked Vault for treasury capital deployment
```
not locked variant, could be Kalani vault

qGolem reviewed (changes requested):
good draft, need to beef up the transaction deploy process to do it all in one go also deploy our morpho strategy to set the donation address and payment splitter (this deposits into the same Morpho strategy IIRC)

0xferit committed:
refactor(shutter): address PR review feedback from qGolem

0xferit committed:
test(shutter): add gas profiling for batched DAO proposal

0xferit committed:
fix(shutter): use Treasury as roleManager per documentation

0xferit committed:
docs: rename Integration Guide to Integration Plan

0xferit committed:
test: add mainnet fork integration test for Shutter DAO

0xferit committed:
test(shutter): replace mock test with mainnet fork integration test

0xferit committed:
fix(test): use ERC1967Proxy for PaymentSplitter deployment

0xferit committed:
docs: update capital to 1.2M and yield split to 0/100

0xferit committed:
docs: finalize yield assumptions and fix comments

0xferit committed:
fix(test): resolve compilation errors and stack depth issues

0xferit committed:
chore: remove stale test file causing CI failure

0xferit committed:
fix(test): remove zero-share payee from PaymentSplitter setup to prevent revert

0xferit commented:
@codex review

0xferit committed:
fix(test): remove unused import to satisfy deny_warnings

0xferit committed:
docs(shutter)!: fix payment splitter revert, add robust delegation tests, and update gas limits per EIP-7825

0xferit committed:
fix(shutter): align tests with source of truth using factory deployment

0xferit committed:
docs(shutter): add prepared calldata for all operations

0xferit committed:
docs(shutter): add execution call chain diagram (Azorius → Safe → Target)

qGolem commented on test/integration/shutter/ShutterDAOIntegration.t.sol:
```diff
@@ -0,0 +1,508 @@
+        ERC1967Proxy proxy = new ERC1967Proxy(address(paymentSplitterImpl), initData);
+        paymentSplitter = PaymentSplitter(payable(address(proxy)));
+
+        // Deploy Morpho Strategy via Factory (per Quentin's source of truth)
```
clean up comments

0xferit committed:
docs(shutter): add complete calldata for all DAO proposal transactions

0xferit committed:
test(shutter): add realistic Azorius → Safe → Target call chain

0xferit committed:
refactor: migrate common constants to src/constants.sol

0xferit committed:
test(shutter): gas profiling uses realistic Azorius module execution

0xferit committed:
docs(shutter): fix role assignment consistency and update gas estimates

0xferit committed:
docs(test): remove historical reference from NatSpec comment

qGolem commented on test/integration/shutter/ShutterDAOIntegration.t.sol:
```diff
@@ -0,0 +1,637 @@
+        uint256[] memory shares = new uint256[](1);
+        shares[0] = 100;
+
+        PaymentSplitter paymentSplitterImpl = new PaymentSplitter();
```
Payment splitter factory instead

0xferit committed:
refactor(test): use PaymentSplitterFactory instead of direct proxy deployment

0xferit committed:
refactor(test): batch multiple transactions via MultiSend

0xferit committed:
refactor(test): remove MultistrategyVault wrapper for single-strategy use

0xferit committed:
docs: simplify integration guide for direct strategy architecture

0xferit committed:
docs: remove historical comparison in gas note

0xferit committed:
fix(shutter): delegatecall multisend in integration tests

qGolem commented on docs/shutter_dao_integration.md:
```diff
@@ -0,0 +1,535 @@
+| **Keeper** | Dedicated Bot/EOA | **REQUIRED**: Authorized to call `report()`/`tend()` to harvest yields without governance votes |
+| **Emergency Admin** | Shutter DAO Treasury (`0x36bD...32c4`) | Can shutdown the strategy and perform emergency withdrawals |
+
+> **Critical**: The Keeper MUST be a dedicated EOA or bot, NOT the Treasury Safe. Assigning Keeper to Treasury would require a governance vote (72-hour voting + 72-hour execution = 144 hours minimum) for every harvest, creating severe operational bottlenecks that defeat the purpose of automated yield generation.
```
should*

qGolem commented on docs/shutter_dao_integration.md:
```diff
@@ -0,0 +1,535 @@
+|-------------|------------|
+| Dragon Funding Pool | 100% |
+
+### Yield Projections (assuming 5% APY)
```
remove for now, though points for accuracy

qGolem commented on docs/shutter_dao_integration.md:
```diff
@@ -0,0 +1,535 @@
+
+## Part 2: Regen Staker
+
+The Regen Staker allows SHU holders to stake their tokens and direct their staking rewards toward public goods funding, which Octant matches. Rewards are distributed from an external source (e.g., SHUGrantPool Strategy yield).
```
dont say that octant matches, that's not our call

qGolem commented on docs/shutter_dao_integration.md:
```diff
@@ -0,0 +1,535 @@
+
+### Key Features
+
+- **Public Goods Funding**: Stakers allocate their rewards to projects in Octant funding rounds
```
think they would run their own rounds in theory

qGolem commented on docs/shutter_dao_integration.md:
```diff
@@ -0,0 +1,535 @@
+### Key Features
+
+- **Public Goods Funding**: Stakers allocate their rewards to projects in Octant funding rounds
+- **Matched Rewards**: Contributions are matched by Octant's matching pool
```
same as bove

qGolem commented on docs/shutter_dao_integration.md:
```diff
@@ -0,0 +1,535 @@
+- No vault wrapper needed (strategy IS the ERC-4626 vault)
+- Fits easily within 16.7M gas (EIP-7825 per-transaction gas limit)
+
+> **Batching note (MultiSend)**: If you batch any of the following calls via MultiSend, execute MultiSend with `operation=DELEGATECALL` from the Safe (Azorius `execTransactionFromModule(..., operation=1)`). Using CALL makes `msg.sender` the MultiSend contract and will break USDC approvals/deposits.
```
I think safe UI doesn't let you use delegate call anymore cc @skimaharvey take a look at this

qGolem commented on src/constants.sol:
```diff
@@ -3,9 +3,48 @@ pragma solidity ^0.8.25;
+
+// Yearn TokenizedStrategy singleton on Ethereum mainnet
+// Required parameter for deploying strategies via MorphoCompounderStrategyFactory
+address constant YEARN_TOKENIZED_STRATEGY_MAINNET = 0xb27064A2C51b8C5b39A5Bb911AD34DB039C3aB9c;
```
*YIELD_DONATING_TOKENIZED_STRATEGY_MAINNET

qGolem commented on partners/shutter_dao_0x36/test/ShutterDAOIntegration.t.sol:
```diff
@@ -0,0 +1,448 @@
+
+    /// @notice Execute batched transactions via MultiSend through Azorius → Safe
+    function _executeBatchFromModule(bytes memory packedTransactions) internal {
+        bytes memory multiSendData = abi.encodeCall(MultiSendCallOnly.multiSend, (packedTransactions));
```
think there is a cannonical multiSend Contract that we can use from Gnosis

qGolem commented on test/integration/shutter/ShutterDAOIntegration.t.sol:
```diff
@@ -0,0 +1,448 @@
+        );
+
+        // TX 2: BATCHED via MultiSend - Approve + Deposit only (no vault config needed!)
+        bytes memory batchedTxs = abi.encodePacked(
```
add the payment splitter deploy and strategy deploy to the batch

qGolem reviewed (changes requested):
add the payment splitter deploy and strategy deploy to the batch in the final integration test and fix nits on proposal text re. matching

0xferit committed:
refactor(shutter): address review comments for integration guide

0xferit committed:
feat(shutter): batch all 4 ops using CREATE2 address precomputation

0xferit committed:
docs(shutter): add fallback for individual transactions

0xferit committed:
docs(shutter): consolidate keeper guidance to avoid duplication

0xferit committed:
feat(shutter): add calldata generator script for DAO proposal

0xferit committed:
docs(shutter): remove stale AutoAllocate section

0xferit committed:
feat(shutter): add test mode to calldata generator script

- Deploy temp PaymentSplitterFactory when placeholder is address(0)
- Use makeAddr() for Keeper and DragonPool placeholders
- Always precompute CREATE2 addresses and output complete calldata
- Add clear TEST MODE indicator in output

0xferit committed:
fix(shutter): capture strategy address from return value instead of CREATE2 prediction

- CREATE2 prediction fails when local bytecode differs from mainnet factory
- Deploy PaymentSplitter + Strategy separately, capture returned address
- Batch approve + deposit with actual strategy address
- Update gas profiling test to use realistic 3-step approach (~2.1M gas)
- All 16 Shutter tests pass

0xferit committed:
fix(shutter): align test execution path with script and update docs

0xferit committed:
docs(shutter): replace governance params table with link to Decent app

0xferit commented:
@codex review

qGolem commented on docs/shutter_dao_integration.md:
```diff
@@ -0,0 +1,545 @@
+
+| Item | Status | Owner | Notes |
+|------|--------|-------|-------|
+| PaymentSplitter Factory deployment | ⏳ Pending | Octant | Required for strategy deployment |
```
"VITE_PAYMENT_SPLITTER_FACTORY_ADDRESS": "0x5711765E0756B45224fc1FdA1B41ab344682bBcb"

qGolem commented on partners/shutter_dao_0x36/test/ShutterDAOIntegration.t.sol:
```diff
@@ -0,0 +1,546 @@
+        bytes memory multiSendData = abi.encodeCall(MultiSendCallOnly.multiSend, (packedTransactions));
+        // Must DELEGATECALL MultiSend so subcalls execute from the Safe (Treasury) address.
+        vm.prank(AZORIUS_MODULE);
+        bool success = ISafe(SHUTTER_TREASURY).execTransactionFromModule(SAFE_MULTISEND_MAINNET, 0, multiSendData, 1);
```
nice

qGolem reviewed (dismissed):
Change payment splitter deployed address and place the files in /partners/shutterDAO

0xferit committed:
feat(shutter): set PaymentSplitter Factory mainnet address

0xferit committed:
refactor(shutter): move integration files to partners/shutter_dao_0x36
